src/app/catalog/filter/filter.component.ts

```diff
@@ -93,17 +97,27 @@ export class FilterComponent implements OnInit {
   }
 
   searchBrands(event: Event) {
-    const input = (event.target as HTMLInputElement).value;
+    const input = (event.target as HTMLInputElement).value.toLocaleLowerCase();
     this.brands = this.allbrands.filter((data) => data.name.toLowerCase().includes(input));
   }
 
   searchSizes(event: Event) {
-    const input = (event.target as HTMLInputElement).value;
+    const input = (event.target as HTMLInputElement).value.toLocaleLowerCase();
     this.sizes = this.allsizes.filter((data) => data.size.toLowerCase().includes(input));
   }
 
   getColors(): string[] {
     const keys = Object.keys(this.colors);
     return  keys.slice(keys.length / 2);
   }
+
+  filterProductByCategory(id: number): void {
+    const filteredProducts: Observable<ProductCard[]> = this.catalogService.filterProductByCategory(id);
+    this.filterEvent.emit(filteredProducts);
+  }
+
+  filterProductBySubCategory(id: number): void {
+    const filteredProducts: Observable<ProductCard[]> = this.catalogService.filterProductBySubCategory(id);
```

Comment:
Since this is observable, emit calls should go inside subscribe method.
BTW is it working like this?

Reply:
Got it, since the product card is expecting observable, it will work.

Reply:
type of productcards is observable